microk8s-resources/actions/common/utils.sh

```diff
@@ -472,7 +472,13 @@ render_csr_conf() {
         local ips='' sep=''
         local -i i=3
         for IP_ADDR in $(echo "$IP_ADDRESSES"); do
-            ips+="${sep}IP.$((i++)) = ${IP_ADDR}"
+            local ip_id="IP.$((i++))"
+            while ! [ '0' == "$(grep -c \"^${ip_id}\" ${SNAP_DATA}/certs/csr.conf.template)" ]
+            do
+              ip_id="IP.$((i++))"
+              echo ${ip_id}
+            done
```

Comment:
Unfortunately this does not work for me, and I think it is related to the way the shell escapes the `^IP.$x` regex.

```suggestion
                while grep '^'"${ip_id}" ${SNAP_DATA}/certs/csr.conf.template > /dev/null ; do
                        ip_id="IP.$((i++))"
                done
```

Reply:
Here grep is not set to used regexp otherwise `.` in the IP would be a problem for IP.3 IP.33 will match also. According to grep documentation it supports ^ and $ without switching to regexp matching. But anyway looks like I pushed the wrong commit will double check.
Actually if in you setup grep is replaced with egrep to enable regexp by default this could be the case.

Reply:
good catch that `IP.33` would also match, but I do not think this is an issue (we might simply skip a few possible values).